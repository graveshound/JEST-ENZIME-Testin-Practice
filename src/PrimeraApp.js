//import React, { Fragment } from 'react';
//Funcional Components
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo2, subtitulo})=>{

    
  
    const saludo="Hola Mundo"

    /*if (!saludo){
        throw new Error("El Salud es Necesario")
    }*/

    return  (
   /* <Fragment>
        <h1>Hola Mundo</h1>
        <p>Primer Parrafo</p>
    </Fragment>
   */
    <>
        <h1> {saludo} </h1>
        <h1> {saludo2} </h1>
        {/*<pre>{JSON.stringify(objeto,null,3)}</pre>*/}
        <p>{subtitulo}</p>
    </>
    );
}
PrimeraApp.propTypes={
    saludo2: PropTypes.string.isRequired

}
PrimeraApp.defaultProps={
    subtitulo:"Soy un sub"
}
export default PrimeraApp;