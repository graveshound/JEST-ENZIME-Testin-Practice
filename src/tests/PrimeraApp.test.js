//import { render } from "@testing-library/react"
import PrimeraApp from "../PrimeraApp"
import React  from 'react';
import {shallow} from 'enzyme'
import '@testing-library/jest-dom'

describe('Prueba de PrimeraAPP', () => {
    //test('debe de mostrar el mensaje "Hola, soy Goku"', () => {
        
        //const saludo="Hola Soy Goku"

        //const { getByText }= render( <PrimeraApp saludo2={saludo} />)

       // expect( getByText(saludo) ).toBeInTheDocument();
   // })

   test('Debe Renderizar correctamente <PrimeraApp></PrimeraApp>', () => {
       
        const saludo="Hola Soy Goku"
        const wrapper=shallow(<PrimeraApp saludo2={ saludo }/>)

        expect(wrapper).toMatchSnapshot();


   })
   
   test('Debe mostrar el subtitulo enviado por props', () => {
    const saludo="Hola Soy Goku"
    const subtitulo="Soy un sub"

    const wrapper=shallow(<PrimeraApp 
        saludo2={ saludo }
        subtitulo={ subtitulo }/>)

    const textoParrafo=wrapper.find('p').text();
    console.log(textoParrafo)

    expect(textoParrafo).toBe(subtitulo);
    
   })
   
    
})
