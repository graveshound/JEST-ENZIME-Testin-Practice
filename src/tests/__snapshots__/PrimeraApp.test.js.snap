// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de PrimeraAPP Debe Renderizar correctamente <PrimeraApp></PrimeraApp> 1`] = `
<Fragment>
  <h1>
     
    Hola Mundo
     
  </h1>
  <h1>
     
    Hola Soy Goku
     
  </h1>
  <p>
    Soy un sub
  </p>
</Fragment>
`;

exports[`Prueba de PrimeraAPP Debe mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
     
    Hola Mundo
     
  </h1>
  <h1>
     
    Hola Soy Goku
     
  </h1>
  <p>
    Soy un sub
  </p>
</Fragment>
`;
